website/blog/2024-06-26-llm-as-judge/index.md

```diff
@@ -0,0 +1,522 @@
+
+2. **[mlflow.metrics.genai.answer_correctness()](https://mlflow.org/docs/latest/python_api/mlflow.metrics.html?highlight=genai%20answer#mlflow.metrics.genai.answer_correctness)**
+
+
+3. **[mlflow.metrics.genai.answer_relevance()](https://mlflow.org/docs/latest/python_api/mlflow.metrics.html?highlight=genai%20answer#mlflow.metrics.genai.answer_relevance)**
+ 
+4. **[mlflow.metrics.genai.relevance()](https://mlflow.org/docs/latest/python_api/mlflow.metrics.html?highlight=genai%20answer#mlflow.metrics.genai.relevance)**
+
+5. **[mlflow.metrics.genai.faithfulness()](https://mlflow.org/docs/latest/python_api/mlflow.metrics.html?highlight=genai%20answer#mlflow.metrics.genai.faithfulness)**
+
+Let's use mlflow's pre-built in faithfulness metric.
+
+
+```python
+from mlflow.metrics.genai import EvaluationExample, faithfulness
+faithfullness_metric = faithfulness(model="openai:/gpt-4")
+print(faithfullness_metric)
+```
+
+This metric works quite well in singery with langchain retrivals since you can provide the grading context seperatly from the llm_input column if you prefer.
+
+Since in this example we are doing everything in the same input column let's map out the context column to our input column.
+
+
+
+```python
+with mlflow.start_run() as run:
+    results = mlflow.evaluate(
+        my_llm,
+        eval_data,
+        model_type="text",
+        evaluators="default",
+        extra_metrics=[faithfullness_metric],
+        evaluator_config={
+        "col_mapping": {
+            "inputs": "llm_inputs",
+            "context": "llm_inputs",
+           }}
+    )
+mlflow.end_run()
+```
+
+![Gauge Faithfullness Chart](faithfulness.png)
+
+
+## Conclusion
+
+MLflow evaluate, combined with LLMs as judges, opens up new possibilities for nuanced and context-aware model evaluation. By creating custom metrics tailored to specific aspects of model performance, data scientists can gain deeper insights into their models' strengths and weaknesses.
+
+The flexibility offered by `make_genai_metric()` allows you to create evaluation criteria that are perfectly suited to your specific use case. Whether you need structured guidance for your LLM judge or want full control over the prompting process, MLflow provides the tools you need.
+
+As you explore MLflow evaluate and LLM-based metrics, remember that the key lies in designing thoughtful evaluation criteria and providing clear instructions to your LLM judge. With these tools at your disposal, you're well-equipped to take your model evaluation to the next level, ensuring that your language models not only perform well on traditional metrics but also meet the nuanced requirements of real-world applications.
+
+
+The built-in metrics, such as toxicity, offer standardized assessments that are crucial for ensuring the safety and accessibility of model outputs. 
```

Comment:
General comments:
- Thanks again for the submission! The llm as judge usage examples will be helpful to readers
- Please give this a close read for spelling/grammar/punctuation/etc
- Include links to the docs where relevant/where you refer to particular features/functions/etc for the first time. In cases where you give details of functionality that is not used in the post, you can instead refer to it briefly and link to the relevant docs.
- @BenWilson2 may also have review comments to add

Reply:
Yep! Left a few general comments. I love the premise and the story here. Let's double-down on the translation evaluation use case and show the story more clearly with demonstrating the translation information and talk a bit more about why having large-scale evaluation of cultural terms is highly relevant when performing advanced translations.

Reply:
Hey, thank you both!

I've added a bit more text on how this is relevant to the example business and how this would work to improve their product if this was a real life scenario. As all as the changes requested.

I've tried to simplify a bit where I could. Let me know if aligns with your vision